apps/web/src/providers/connectionConfig/indexedDbRepository.ts

```
@@ -0,0 +1,129 @@
+        const localRepositoryConnections = await localRepository.list();
+
+        // Check if data from localRepository should be migrated to indexedDbRepository
+        if (
+            indexedDbConnections.length === 0 &&
+            localRepositoryConnections.length > 0
+        ) {
+            const connections = localRepositoryConnections.map(
+                (connection) => ({
+                    ...connection,
+                    network: networkId,
+                }),
+            ) as ConnectionItem[];
+
+            // Save augmented connections
+            await this.connections.bulkPut(connections);
+
+            // Remove connections from localRepository to keep a single source of truth
+            await Promise.all(
+                connections.map((c) => localRepository.remove(c.address)),
+            );
+        }
+    }
+
+    private formatConnection(connection: ConnectionItem) {
+        return {
+            address: connection.address,
+            url: connection.url,
+            timestamp: connection.timestamp,
+        } as Connection;
+    }
+
+    async add(conn: Connection) {
+        const connectionItem: ConnectionItem = {
+            ...conn,
+            timestamp: Date.now(),
+            network: networkId,
+        };
+
+        return this.connections.add(connectionItem);
+    }
+
+    async has(addr: Address) {
+        const connection = await this.connections
+            .where("network")
+            .equals(networkId)
+            .and((connection) => connection.address === addr)
+            .first();
+
+        return Boolean(connection);
+    }
+
+    async remove(addr: Address) {
+        try {
+            await this.connections
+                .where("network")
+                .equals(networkId)
+                .and((connection) => connection.address === addr)
+                .delete();
+            return true;
+        } catch (err) {
+            return false;
```

Hi @nevendyulgerov, I would not catch and swallow the error here because it is up to who is calling the `remove` method to catch and decide what to do with it. That is the case for the "connection-hooks" and the `lifecycle` actions where we catch and log the error reason and also pass up in case the hook's caller attached a callback interested in listening to failures to do something in the UI (e.g. Notification).


Here for remove: https://github.com/cartesi/rollups-explorer/blob/f321db064bc47c6bfff9c6e88224c09e5546c308/apps/web/src/providers/connectionConfig/hooks.tsx#L91
Here for add: https://github.com/cartesi/rollups-explorer/blob/f321db064bc47c6bfff9c6e88224c09e5546c308/apps/web/src/providers/connectionConfig/hooks.tsx#L71